icu4c/source/i18n/tridpars.cpp

```diff
@@ -136,6 +136,9 @@ TransliteratorIDParser::parseSingleID(const UnicodeString& id, int32_t& pos,
                 specsB = parseFilterID(id, pos, TRUE);
                 // Must close with a ')'
                 if (specsB == NULL || !ICU_Utility::parseChar(id, pos, CLOSE_REV)) {
+                    if (specsB != NULL) {
```

Comment:
could we change 
```
Specs* specsA
Specs* specsB
```
to
```
LocalMemory<Specs> specsA;
LocalMemory<Specs> specsB;
```
to fix this?
@markusicu  any opinion?

Reply:
smart pointers are always safer

in this case, the change looks ok; i would actually `delete specsB` unconditionally

Reply:
@FrankYFTang @markusicu like this?:

```diff
--- a/icu4c/source/i18n/tridpars.cpp
+++ b/icu4c/source/i18n/tridpars.cpp
@@ -136,6 +136,7 @@ TransliteratorIDParser::parseSingleID(const UnicodeString& id, int32_t& pos,
                 specsB = parseFilterID(id, pos, true);
                 // Must close with a ')'
                 if (specsB == nullptr || !ICU_Utility::parseChar(id, pos, CLOSE_REV)) {
+                    delete specsB;
                     delete specsA;
                     pos = start;
                     return nullptr;
```